### PR TITLE
bib: add a bunch of DOIs, minor corrections

### PR DESCRIPTION
**Pull request by fingolfin**
DOIs added via script from <https://tex.stackexchange.com/a/300474>. The DOIs were then checked with <https://www.doi2bib.org> and in a few cases further adjustments were made to the bib data.

Some progress towards https://github.com/oscar-system/Oscar.jl/issues/4214.

### CODE REVIEW

File: docs/oscar_references.bib [hunk 1]
```diff
@@ -206,13 +207,17 @@ @InProceedings{BES-E-D21
...
-  eprint        = {1905.07114},
-  archiveprefix = {arXiv},
-  primaryclass  = {math.CO}
+  journal       = {JEMS},
+  fjournal      = {Journal of the European Mathematical Society},
+  volume        = {26},
+  number        = {11},
+  pages         = {4491--4535},
+  year          = {2023},
+  month         = jun,
```
Comment (lgoettgens) [inline docs/oscar_references.bib:219]:
```suggestion
```
we usually don't keep track of months, and here it is formatted differently as all other fields

File: docs/oscar_references.bib [hunk 2]
```diff
@@ -249,13 +254,15 @@ @Article{BH23
...
   title         = {A semi-small decomposition of the Chow ring of a matroid},
-  year          = {2020},
-  eprint        = {2002.03341},
-  archiveprefix = {arXiv},
-  primaryclass  = {math.AG}
+  journal       = {Advances in Mathematics},
+  volume        = {409},
+  pages         = {108646},
+  year          = {2022},
+  month         = nov,
```
Comment (lgoettgens) [inline docs/oscar_references.bib:264]:
```suggestion
```

**Review by lgoettgens — APPROVED** [review]
Thanks! Two minor comments

File: docs/oscar_references.bib (hunk 1 above)
Comment (joschmitt) [reply docs/oscar_references.bib:219]: If we usually don't keep track of months, shouldn't bibtool be configured to delete the fields?

Comment (lgoettgens) [reply docs/oscar_references.bib:219]: There are months in the bibliography, but there is a wild mix of formatting already in there:
```
  month         = {Mar},
  month         = {1},
  month         = jun,
  month         = {mar},
```
Thus I would keep this PR as is and instead convert this to a new issue that deals with unifying the month fields or deciding to delete them (both using appropriate bibtool settings).

**Pull request merged**